azerothcore/azerothcore-wotlk
# fix(DB): Removes multiple occurances of double ore spawn

## Changes Proposed:
- Adds pools to ores which have the same location of these types: copper(1731, 2055, 3763, 103713, 181248),tin(1732), silver(1733), gold(1734), iron(1735), mithril(2040), truesilver(2047, 150081), dark iron(165658), small thorium(324, 150082, 176643), rich thorium(175404), fel iron(181555), adamantite(181556), rich adamantite(181569), cobalt(189978), rich cobalt(189979), khorium(181557), saronite(189980), rich saronite(189981) & titanium(191133).
- These pools make it so that if multiple ores (of this set) have the same location, only one of them can spawn at a time. 
- Limits the amount (of this set) that can spawn in a zone. (I did not have a reference amount, so just set it to half rounded down)

Where 'this set' is mentioned, the ores of same location but no pool is meant.

## Issues Addressed:

- Closes #5735
- Closes https://github.com/chromiecraft/chromiecraft/issues/572
- Closes #5742
- Closes https://github.com/chromiecraft/chromiecraft/issues/578
- Closes #5789
- Closes https://github.com/chromiecraft/chromiecraft/issues/590
- Closes #5883
- Closes https://github.com/chromiecraft/chromiecraft/issues/654
- Closes #5904
- Closes https://github.com/chromiecraft/chromiecraft/issues/660
- Closes #5998
- Closes #6215
- Closes https://github.com/chromiecraft/chromiecraft/issues/769
- Closes #6947
- Closes https://github.com/chromiecraft/chromiecraft/issues/1199
- Closes #6603
- Closes https://github.com/chromiecraft/chromiecraft/issues/1012

## SOURCE:

There are multiple changes in this pull request. 
* Preventing multiple ores from spawning in the same place - This I don't believe is controversial
* Limiting spawns in a zone of the above ores - This could be wrong, but having some kind of limit is proberly correct
* The spawn points, locations of multiple ores, have a chance of spawning each ore - I honestly have no idea here. I did try to find a reference to another spawn point where the same types of ores spawned, but I could be wrong.



## Tests Performed:

As described below.


## How to Test the Changes:


1. Run SQL commands
2. Check correct amount of rows changed.
- Should delete 0 rows
- Should insert 54 rows into \`pool_templates\`, 36 rows into \`pool_pool\`, and 74 rows into \`pool_gameobject\`

3. Check that each zone, as described below, has the correct amount of spawns.
4. Check that each spawn point has at most one spawned ore. Note: A spawn point can have no ores spawned.

Zones:
Redridge Mountains - 1 spawned ore
Spawn point 1 (Guids): 5656   / 5749

Stranglethorn Vale - 1 spawned ore
Spawn point 1 (Guids): 5879   / 6470
Spawn point 2 (Guids): 63539 / 65223
Spawn point 3 (Guids): 40019 / 6187

Thousand Needles - 1 spawned ore
Spawn point 1 (Guids): 6524   / 17159

Searing Gorge - 4 spawned ore
Spawn point 1 (Guids): 5850   / 7234   / 9535 / 58117
Spawn point 2 (Guids): 33606 / 34816
Spawn point 3 (Guids): 10004 / 35299
Spawn point 4 (Guids): 373     / 10214
Spawn point 5 (Guids): 10007 / 64842
Spawn point 6 (Guids): 10103 / 56457
Spawn point 7 (Guids): 10165 / 64838
Spawn point 8 (Guids): 258     / 56819

Uldaman - 1 spawned ore
Spawn point 1 (Guids): 13024 / 33204

Desolace- 1 spawned ore
Spawn point 1 (Guids): 29207 / 63514

Darkshore - 1 spawned ore
Spawn point 1 (Guids): 30160 / 48554

Silverpine Forest - 1 spawned ore
Spawn point 1 (Guids): 35547 / 42971

Blackfathom Deeps - 1 spawned ore
Spawn point 1 (Guids): 48120 / 63507

Tanaris - 1 spawned ore
Spawn point 1 (Guids): 5999   / 65300

Maraudon - 1 spawned ore
Spawn point 1 (Guids): 7232   / 9239
Spawn point 2 (Guids): 7180   / 9406
Spawn point 3 (Guids): 63965 / 32922

Winterspring - 1 spawned ore
Spawn point 1 (Guids): 9445   / 18433

Hellfire Peninsula - 1 spawned ore
Spawn point 1 (Guids): 42301 / 61963
Spawn point 2 (Guids): 64873 / 64885

Nagrand - 2 spawned ore
Spawn point 1 (Guids): 21915 / 61931
Spawn point 2 (Guids): 21924 / 61927
Spawn point 3 (Guids): 40200 / 61919
Spawn point 4 (Guids): 42414 / 42431
Spawn point 5 (Guids): 61935 / 63206

Terokkar Forest - 1 spawned ore
Spawn point 1 (Guids): 40271 / 61962
Spawn point 2 (Guids): 14124 / 61966
Spawn point 3 (Guids): 28489 / 64887

Mana-Tombs - 1 spawned ore
Spawn point 1 (Guids): 43147 / 63197

The Underbog - 1 spawned ore
Spawn point 1 (Guids): 61906 / 63200

Zangarmarsh - 1 spawned ore
Spawn point 1 (Guids): 34924 / 64884

## Known Issues and TODO List:


- The ores in Blackfathom Deeps are out of bounds.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
